<a href="https://colab.research.google.com/github/richardp123456/PersonalProjects/blob/main/Pilz_R_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

For this data analysis with python, the daily units sold are correlated wit the temperature. The higher the temperature, the higher the units sold per day. However, due to the positive correlation between the products themselves, it can be hypothesized that the higher sales stem from the seasonal climate changes in Australian, than the products themselves.

The File_path is through the google collaboration notebook. I uploaded the csv file into the notebook.
I will skip the first 2 rows by using the skiprows function.

In [102]:
import pandas as pd
import os

file_path = '/content/daily_sales.csv'
df = pd.read_csv(file_path, skiprows = 2)
df.head()


Date  Daily Units Sold  Daily Unit Price  Daily Temperature (C)
0  1/1/2024                91              24.0                     25
1  1/2/2024                90              24.0                     24
2  1/3/2024                70              24.0                     19
3  1/4/2024                89              24.0                     23
4  1/5/2024               100              24.0                     36

The describe function gives an opportunity to see a sumarized table with count, mean, std, min, 25%, 50%, 75% and max.

In [103]:
df.describe().round()

Daily Units Sold  Daily Unit Price  Daily Temperature (C)
count             244.0             244.0                  244.0
mean               66.0              22.0                   20.0
std                19.0               1.0                    5.0
min                11.0              20.0                    7.0
25%                57.0              21.0                   16.0
50%                64.0              22.0                   19.0
75%                81.0              22.0                   23.0
max               100.0              24.0                   36.0

**Interpretation**

The Count column means that there are 244 instances of none null values in each column minus the 2 skipped rows.
The mean is the average of all values in each respective column.
The std shows the standard deviation in each column from the mean.
The min shows the minimum values in each respective column.
The 25%, 50% and the 75% gives a benchmark of what values lie under the respective percentages of the data.
the max shows the maximum value in each column.

In [105]:
from scipy.stats import pearsonr

coeff = pearsonr(df['Daily Units Sold'], df['Daily Temperature (C)'])
print(f"\nPearson correlation coefficient between Daily Units Sold and Temperature: {coeff}")




Pearson correlation coefficient between Daily Units Sold and Temperature: PearsonRResult(statistic=0.8044269923525683, pvalue=1.171028703680638e-56)


I changed the newly sorted csv file name by usig the os.rename function.

In [111]:
df_sorted = df.sort_values(by='Daily Units Sold')
df_sorted.to_csv('daily_sales_sorted.csv', index=False)
new_file_name = 'WaterCure_analysis_Pilz_Richard.csv'
os.rename('daily_sales_sorted.csv', 'WaterCure_analysis_Pilz_Richard.csv')
from google.colab import files
files.download(new_file_name)

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>